python_modules/dagster-graphql/dagster_graphql/implementation/execution/run_lifecycle.py

```diff
@@ -39,7 +40,20 @@ def is_resume_retry(execution_params):
 
 
 def create_valid_pipeline_run(graphene_info, external_pipeline, execution_params):
-    ensure_valid_config(external_pipeline, execution_params.mode, execution_params.run_config)
+    if execution_params.mode is None and len(external_pipeline.available_modes) > 1:
+        raise DagsterInvariantViolationError(
+            f"Pipeline {external_pipeline.name} has multiple modes (Available modes: "
+            f"{external_pipeline.avaliable_modes}) and you have "
```

Comment:
Typo here: `avaliable_modes`. Maybe worth testing this codepath to ensure that a `DagsterInvariantViolationError` gets thrown?

Reply:
yea good point.